src/vs/workbench/services/editor/common/customEditorLabelService.ts

```diff
@@ -139,7 +139,7 @@ export class CustomEditorLabelService extends Disposable implements ICustomEdito
 		for (const pattern of this.patterns) {
 			let relevantPath: string;
 			if (root && !pattern.isAbsolutePath) {
-				relevantPath = relativePath ?? getRelativePath(resourceDirname(root.uri), resource) ?? resource.path;
+				relevantPath = relativePath = relativePath ?? getRelativePath(resourceDirname(root.uri), resource) ?? resource.path;
```

Comment:
I don't think this does anything? If it does you should change it to be more obvious about what it's doing

Reply:
It does, it caches the value. I'll make it a bit more readable to others